Feat/kevin 1224

kevin-hashimoto opened the pull request:
Templates for email notification based on the notification type triggers from #1314
**note**: this does not include implementation as that will be included in #1126 

- feat: email notification templates, closes #1224

considerations: many of the templates are currently the same. we can consolidate these into a single template for simplicity unless we will be needing different wording/url slugs.

hamed-valiollahi commented on backend/lcfs/web/api/email/template_mapping.py:
```diff
@@ -1,20 +1,20 @@
+    "IDIR_DIRECTOR__COMPLIANCE_REPORT__MANAGER_RECOMMENDATION": "idir_director__compliance_report__manager_recommendation.jinja2",
+    "IDIR_DIRECTOR__INITIATIVE_AGREEMENT__ANALYST_RECOMMENDATION": "idir_director__initiative_agreement__analyst_recommendation.jinja2",
+    "IDIR_DIRECTOR__TRANSFER__ANALYST_RECOMMENDATION": "idir_director__transfer__analyst_recommendation.jinja2",
     "default": "default.html",
```
Do we need a Jinja2 extension for default as well?

hamed-valiollahi reviewed (approved):
Looks great! 👍 I added one minor comment. Not really sure if the default template is necessary.

kevin-hashimoto commented:
@hamed-valiollahi good call, i removed the unused template files. I also simplified the template files to just pass variables as the content was all the same. If you could take another look that would be great! thanks!
